src/main/java/com/stablekernel/standardlib/LoadingImageView.java

```diff
@@ -0,0 +1,70 @@
+package com.stablekernel.standardlib;
+
+import android.content.Context;
+import android.content.res.TypedArray;
+import android.graphics.drawable.Drawable;
+import android.support.annotation.AttrRes;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.util.AttributeSet;
+import android.util.TypedValue;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.FrameLayout;
+import android.widget.ImageView;
+import android.widget.ProgressBar;
+
+import static android.view.ViewGroup.LayoutParams.WRAP_CONTENT;
+
+public class LoadingImageView extends FrameLayout {
+    private ImageView imageView;
+    private ProgressBar progressBar;
+
+    public LoadingImageView(@NonNull Context context) {
+        this(context, null);
+    }
+
+    public LoadingImageView(@NonNull Context context, @Nullable AttributeSet attrs) {
+        this(context, attrs, 0);
+    }
+
+    public LoadingImageView(@NonNull Context context, @Nullable AttributeSet attrs, @AttrRes int defStyleAttr) {
+        super(context, attrs, defStyleAttr);
+
+        View view = LayoutInflater.from(context).inflate(R.layout.loading_image_view, this, true);
+        imageView = (ImageView) view.findViewById(R.id.imageView);
+        progressBar = (ProgressBar) view.findViewById(R.id.progressBar);
+
+        final TypedArray a = context.obtainStyledAttributes(attrs, R.styleable.LoadingImageView);
+
+        Drawable image = a.getDrawable(R.styleable.LoadingImageView_imageSrc);
+        int imageHeight = a.getDimensionPixelSize(R.styleable.LoadingImageView_imageHeight, dipsToPix(WRAP_CONTENT));
+        int imageWidth = a.getDimensionPixelSize(R.styleable.LoadingImageView_imageWidth, dipsToPix(WRAP_CONTENT));
+
+        imageView.setImageDrawable(image);
+        imageView.getLayoutParams().height = imageHeight;
+        imageView.getLayoutParams().width = imageWidth;
+
+        a.recycle();
+    }
+
+    public void setImageResource(int resourceId) {
+        imageView.setImageResource(resourceId);
+    }
+
+    public void setImageDrawable(Drawable drawable) {
+        imageView.setImageDrawable(drawable);
+    }
+
+    public void setProgressVisibility(int visibility) {
```

Comment:
If you wanted to get real slick here, you could think to how people will use this. Either they want to show progress or they don't. So instead of forcing your user to input a visibility, you could take in a boolean and then switch visible <----> gone.

This would also allow you to ensure the right thing was done: the progress indicator was set to GONE instead of INVISIBLE.